sound/soc/sof/core.c

```diff
@@ -446,22 +446,42 @@ static void sof_probe_work(struct work_struct *work)
 		/* errors cannot be propagated, log */
 		dev_err(sdev->dev, "error: %s failed err: %d\n", __func__, ret);
 	}
+
+	/* enable runtime_pm */
+	pm_runtime_set_autosuspend_delay(sdev->dev, SND_SOF_SUSPEND_DELAY_MS);
+	pm_runtime_use_autosuspend(sdev->dev);
+
+	pm_runtime_enable(sdev->dev);
+
+	/* mark last_busy for pm_runtime to make sure not suspend immediately */
+	pm_runtime_mark_last_busy(sdev->dev);
+
+	pm_runtime_put_noidle(sdev->dev);
```

Comment:
this needs to be in probe_continue, otherwise PM will not be enabled with you don't have a work queue.

You also need to check if you  need to enable pm_runtime in the parent first or not? No sure what the recommended sequence is.